### PR TITLE
Add CHART.JS v4 compatibility

### LINKED ISSUE #164: Upgrading Chart.JS to version 4.x,  treemap throws error
There's a new Chart.JS version 4x. Upgrading that, I get the following error from treemap-chart

Screenshot 2022-11-15 at 13 56 50

My line 11
```
    Chart.register(TreemapController, TreemapElement);
```

### PR DESCRIPTION
**Pull request by stockiNail**
Fix #164

 - [x] solve a couple of TS types issues.
 - [x] chartjs-adapter-date-fns not compatible with CHART-JS 4 yet.
 - [ ] chart-plugin-datalabels not compatible with CHART-JS 4 yet

### CODE REVIEW

**Comment by stockiNail** [comment]
@kurkle I'm going on tomorrow. About `requireVersion` function, it didn't work with version 4.0.1. I have copied the implementation of annotation plugin, so they are aligned.

**Comment by kurkle** [comment]
I think it only needs updated date-fns adapter and should be fine.

**Comment by stockiNail** [comment]
I have an issue importing color functions from the helpers of Chartjs in the types test. I wrote it in slack.
Sounds an issue of version 4 but not sure

**Comment by stockiNail** [comment]
> I think it only needs updated date-fns adapter and should be fine.

having a look to the error, it seems more related to [datalabels](https://github.com/kurkle/chartjs-chart-treemap/actions/runs/3499080455/jobs/5860225532). Date-fns adapter version updated to 2.0.1.

**Comment by stockiNail** [comment]
> I have an issue importing color functions from the helpers of Chartjs in the types test. I wrote it in slack.
> Sounds an issue of version 4 but not sure

I have changed tsconfig and now it's working

**Comment by stockiNail** [comment]
@kurkle do you know why date-fns is needed? I don't see any place where it's used (if I'm not wrong)

File: types/tests/tsconfig.json [hunk 1]
```diff
@@ -8,6 +8,6 @@
   },
   "include": [
     "../index.esm.d.ts",
-    "./**/*.ts",
+    "../../dist/**/*.d.ts"
```
Comment (kurkle) [inline types/tests/tsconfig.json:11]: Why this change?
If I understand correctly it skips all the ts tests now.

Comment (kurkle) [reply types/tests/tsconfig.json]: I'm guessing this is an issue with typescript module resolution.
Maybe `"moduleResolution": "NodeNext",` in tsconfig would help? (could also bring other issues)

Comment (stockiNail) [reply types/tests/tsconfig.json]: I have tried right now.. It doesn't help.

```
npm run test-types

> chartjs-chart-treemap@2.1.3 test-types
> tsc -p types/tests/

node_modules/chart.js/dist/helpers.d.ts:1:23 - error TS2307: Cannot find module '@kurkle/color' or its corresponding type declarations.

1 import { Color } from '@kurkle/color';
                        ~~~~~~~~~~~~~~~

types/index.esm.d.ts:8:8 - error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("chart.js")' call instead.

8 } from 'chart.js';
         ~~~~~~~~~~

types/index.esm.d.ts:111:16 - error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("chart.js")' call instead.

111 declare module 'chart.js' {
                   ~~~~~~~~~~

types/tests/options.ts:2:23 - error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("chart.js")' call instead.
  To convert this file to an ECMAScript module, change its file extension to '.mts', or add the field `"type": "module"` to 'C:/Users/AndreaStocchero/git/chartjs-chart-treemap/package.json'.

2 import { Chart } from 'chart.js';
                        ~~~~~~~~~~

types/tests/options.ts:3:35 - error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("chart.js/helpers")' call instead.
  To convert this file to an ECMAScript module, change its file extension to '.mts', or add the field `"type": "module"` to '/git/chartjs-chart-treemap/package.json'.

3 import { color as colorLib } from 'chart.js/helpers';
                                    ~~~~~~~~~~~~~~~~~~


Found 5 errors in 3 files.

Errors  Files
     1  node_modules/chart.js/dist/helpers.d.ts:1
     2  types/index.esm.d.ts:8
     2  types/tests/options.ts:2

```

Comment (stockiNail) [reply types/tests/tsconfig.json]: @kurkle I didn't find a way...

Nevertheless, in version 3 of CHART.JS was working because the types didn't expose the kurkle/color lib but the types were wrapped: https://cdn.jsdelivr.net/npm/chart.js@3.9.1/types/helpers/helpers.color.d.ts

Now, having the explicit import, I think (but not sure) you must add the dependency.

Comment (stockiNail) [reply types/tests/tsconfig.json]: what I cannot understand is why is working on a normal TS project:

https://codesandbox.io/s/typescript-forked-fb4e3s?file=/src/index.ts

Comment (kurkle) [reply types/tests/tsconfig.json]: I guess the error is masked:
image

compared to this repo with @kurkle/color added as devDependency:
image

Comment (kurkle) [reply types/tests/tsconfig.json]: So, as a workaround, I think its ok to add `@kurkle/color` as devDependency here, until the issue is solved in chart.js.

Comment (kurkle) [reply types/tests/tsconfig.json]: Or: https://github.com/chartjs/chartjs-plugin-zoom/pull/695/files/fcf885f159b2be3212b7a6cb03689820b51aea46..be9a095f957266f40a9495cfbae6aa840f02b443

Comment (stockiNail) [reply types/tests/tsconfig.json]: Believe me I was having a look to that commit in zoom. Maybe its better to try this parameter.
But just for my knowledge, do you think its something to fix in Chartjs?

Comment (kurkle) [reply types/tests/tsconfig.json]: yes, I think so. It refers to a package for types and does not require that package to be installed. it either needs to bundle the types, not export them or require the dependency to be a nice citizen.

Comment (stockiNail) [reply types/tests/tsconfig.json]: Great!! Tomorrow I will apply the new parameter

Comment (stockiNail) [reply types/tests/tsconfig.json]: Done! Thank you @kurkle

Comment (kurkle) [reply types/tests/tsconfig.json]: @stockiNail sure, go ahead and submit the issue (I was too lazy to do that)
